src/librustc_mir/borrow_check/error_reporting.rs

```diff
@@ -182,13 +182,23 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
             );
 
             if !is_loop_move {
-                err.span_label(
-                    span,
-                    format!(
-                        "value {} here after move",
-                        desired_action.as_verb_in_past_tense()
-                    ),
-                );
+                if used_place.is_prefix_of(&moved_place) {
```

Comment:
You should be checking that if, for any of the `MoveSite` in `move_site_vec`, the `Place` associated with that `MoveSite` is a prefix of `used_place`. I describe how to get the `Place` for a `MoveSite` in the mentor instructions.